tests/test_uniswapv2.py

```diff
@@ -0,0 +1,148 @@
+        self.token_b = Token("USDC", Address("0x198419c5c340e8de47ce4c0e4711a03664d42cb2"), 6)
+        self.uniswap = UniswapV2(self.web3, self.token_a, self.token_b)
+
+    def test_get_block(self):
+        assert isinstance(self.uniswap.get_block(), int)
+
+    def test_approval(self):
+        # given
+        assert self.dai.allowance_of(self.our_address, self.router_address) == Wad(0)
+
+        # when
+        # self._router_contract.approve([self.dai], directly())
+        approval_function = directly()
+        
+        approval_function(self.dai, self.router_address, 'IUniswapV2Router02')
+
+        # then
+        assert self.dai.allowance_of(self.our_address, self.router_address) > Wad(0)
+
+    def test_getting_balances(self):
+        # given
+        # self.dgx.mint(Wad(17 * 10**9)).transact()
+        self.uniswap.get_account_token_balance(self.token_a)\
+            # mint(Wad.from_number(17)).transact()
+
+        # when
+        # balances = self.uniswap.get_balances(self.pair)
+        # then
+        # assert balances[0] == Wad.from_number(17)
+        # assert balances[1] == Wad.from_number(17)
+
+    def test_add_liquidity_eth(self):
+
+        eth_pair_amounts = {
+            "amount_token_desired": self.web3.toWei(24, 'ether'),
+            "amount_token_min": self.web3.toWei(21, 'ether'),
+            "amount_eth_min": self.web3.toWei(.1, 'ether')
+        }
+
+        approval_function = directly()
+        approval_function(self.dai, self.router_address, 'IUniswapV2Router02')
+
+        time.sleep(20)
+        # when
+        add_liquidity = self.uniswap.add_liquidity_eth(eth_pair_amounts, self.dai.address).transact()
+
+        # then
+        assert add_liquidity.result == True
+
+        # when
+        # orders = self.uniswap.get_orders(self.pair, [add_liquidity])
+        # # then
+        # assert orders[0].order_id is not place_order
+        # assert orders[0].is_sell is True
+        # assert orders[0].price == Wad.from_number(45.0)
+        # assert orders[0].amount == Wad.from_number(5.0)
+        # assert orders[0].add_liquidity == add_liquidity
+
+    def test_remove_liquidity_eth(self):
+        # when
+        zrx_order = self.uniswap.place_order(self.pair, False, Wad.from_number(45.0), Wad.from_number(5.0), 999)
+        # then
+        assert zrx_order.buy_token == self.dgx.address
+        assert zrx_order.buy_amount == Wad(5 * 10**9)
+        assert zrx_order.pay_token == self.dai.address
+        assert zrx_order.pay_amount == Wad.from_number(5.0 * 45.0)
+        assert zrx_order.expiration == 999
+
+        # when
+        orders = self.uniswap.get_orders(self.pair, [zrx_order])
+        # then
+        assert orders[0].order_id is not None
+        assert orders[0].is_sell is False
+        assert orders[0].price == Wad.from_number(45.0)
+        assert orders[0].amount == Wad.from_number(5.0)
+        assert orders[0].zrx_order == zrx_order
+
+    def test_swap():
```

Comment:
Test is empty, and missing newline at end of file

Reply:
Updated tests